src/osx/slider_osx.cpp

```diff
@@ -11,23 +11,64 @@
 
 #if wxUSE_SLIDER
 
+#include <iostream>
+
 #include "wx/slider.h"
 #include "wx/osx/private.h"
 
 wxBEGIN_EVENT_TABLE(wxSlider, wxControl)
 wxEND_EVENT_TABLE()
 
- // The dimensions of the different styles of sliders (from Aqua document)
+namespace
+{
+    // The dimensions of the different styles of sliders (from Aqua document)
 #if wxOSX_USE_COCOA
-    #define wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS 28
-    #define wxSLIDER_DIMENSIONACROSS_ARROW 21
+    const int wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS = 28;
+    const int wxSLIDER_DIMENSIONACROSS               = 28;
 #else
-    #define wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS 24
-    #define wxSLIDER_DIMENSIONACROSS_ARROW 18
+    const int wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS = 24;
+    const int wxSLIDER_DIMENSIONACROSS               = 18;
 #endif
 
-// Distance between slider and text
-#define wxSLIDER_BORDERTEXT 5
+    const int DISTANCE_SLIDER_TEXT = 5; // Distance between slider and text
+
+    // Returns the best dimension of the slider perpendicular to the slider's moving direction
+    int GetBestSliderAcrossDimension(wxSlider const* slider)
+    {
+        return (((slider->GetWindowStyle() & wxSL_AUTOTICKS) != 0) ?
+                wxSLIDER_DIMENSIONACROSS_WITHTICKMARKS : wxSLIDER_DIMENSIONACROSS);
+    }
+    // Returns the slider's text position perpendicular to the slider's moving direction
+    /*
+     * When the slider's size prependicular to the slider's moving direction becomes
+     * smaller than the best size, the formula for determining the text position is
+     * not valid anymore. Therefor, the text is aligned under these conditions with
+     * respect to the lower (horizontal slider orientation), respectively to the
+     * right (vertical slider orientation) border of the slider.
+     * Remarks: There is still a small glitch in calculating the position, namely when
+     *          wxWidgets' calculation determines a size of zero in the perpendicular
+     *          dimension. In this case the native control's dimension is still not zero
+     *          because wxWidgets adds some insets. So, the slider is still visible
+     *          while wxWidgets thinks it is already invisble. In this case the
+     *          position calculcation is also wrong as it is based on wxWidgets's data.
+     */
+    int GetTextAcrossPosition(wxSlider const* slider, wxSize const& textDimension)
+    {
+        const wxSize size     = slider->GetSize();
+        const wxSize bestSize = slider->GetBestSize();
+
+
+        if (slider->GetWindowStyle() & wxSL_VERTICAL)
+            if (size.x >= bestSize.x)
+                return GetBestSliderAcrossDimension(slider)+DISTANCE_SLIDER_TEXT;
+            else
+                return MAX(0,size.x-textDimension.x);
```

Comment:
What is this `MAX`? It results in build errors and I don't see where it is supposed to be defined... We have `wxMax()` in `wx/utils.h` and there is, of course, also `std::max()` and one of them should be used instead.

Reply:
Sorry, has been taken from a macOS system header. Will use wxMax instead.